pythia/models/butd.py

```diff
@@ -0,0 +1,294 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+
+import gc
+import torch
+from pythia.common.registry import registry
+from pythia.modules.encoders import ImageEncoder
+from pythia.modules.decoders import CaptioningDecoder
+from torch import nn
+
+from .base_model import BaseModel
+
+
+@registry.register_model("butd")
+class BUTD(BaseModel):
```

Comment:
Sounds good. Refactoring this will need retraining the current model we have. I will add these changes after I finish training with changes in this part.